analyzers/src/SonarAnalyzer.Common/Rules/Hotspots/PubliclyWritableDirectoriesBase.cs

```diff
@@ -29,27 +29,26 @@ public abstract class PubliclyWritableDirectoriesBase<TSyntaxKind, TInvocationEx
         protected const string DiagnosticId = "S5443";
         private const string MessageFormat = "Make sure publicly writable directories are used safely here.";
         private const RegexOptions WindowsAndUnixOptions = RegexOptions.Compiled | RegexOptions.IgnoreCase | RegexOptions.CultureInvariant;
-        private readonly string[] linuxDirs =
-        {
-            "/dev/mqueue",
-            "/run/lock",
-            "/var/run/lock",
-        };
-        private readonly string[] macDirs =
+
+        protected static string[] InsecureEnvironmentVariables { get; }
+
+        private static readonly Regex UserProfile;
+        private static readonly Regex LinuxDirectories;
+        private static readonly Regex MacDirectories;
+        private static readonly Regex WindowsDirectories;
+        private static readonly Regex EnvironmentVariables;
+
+        static PubliclyWritableDirectoriesBase()
         {
-            "/var/tmp",
-            "/usr/tmp",
-            "/dev/shm",
-            "/library/caches",
-            "/users/shared",
-            "/private/tmp",
-            "/private/var/tmp",
-        };
-        private readonly Regex userProfile = new(@"^%USERPROFILE%[\\\/]AppData[\\\/]Local[\\\/]Temp", WindowsAndUnixOptions);
-        private readonly Regex linuxDirectories;
-        private readonly Regex macDirectories;
-        private readonly Regex windowsDirectories = new(@"^([a-z]:[\\\/]?|[\\\/][\\\/][^\\\/]+[\\\/]|[\\\/])(windows[\\\/])?te?mp([\\\/]|$)", WindowsAndUnixOptions);
-        private readonly Regex environmentVariables;
+            var insecureEnvironmentVariables = new[] { "tmp", "temp", "tmpdir" };
+            InsecureEnvironmentVariables = insecureEnvironmentVariables;
```

Comment:
Why are you using an intermediate variable for `InsecureEnvironmentVariables`?
```suggestion
            InsecureEnvironmentVariables = new[] { "tmp", "temp", "tmpdir" };
```

Reply:
That way, the property assignments can be safely re-ordered.